src/routes/transactions/entities/swap-order-info.entity.ts

```diff
@@ -7,168 +7,132 @@ import {
-  expiresTimestamp: number;
+  validTo: number;
 
   @ApiProperty({
-    description: 'The filled percentage of the order',
-    examples: ['0.00', '50.75', '100.00'],
+    description: 'The sell token raw amount (no decimals)',
   })
-  filledPercentage: string;
-
-  @ApiProperty({ description: 'The URL to the explorer page of the order' })
-  explorerUrl: URL;
-
-  protected constructor(args: {
-    orderUid: string;
-    status:
-      | 'open'
-      | 'fulfilled'
-      | 'cancelled'
-      | 'expired'
-      | 'presignaturePending';
-    orderKind: 'buy' | 'sell';
-    sellToken: TokenInfo;
-    buyToken: TokenInfo;
-    expiresTimestamp: number;
-    filledPercentage: string;
-    explorerUrl: URL;
-  }) {
-    super(TransactionInfoType.SwapOrder, null, null);
-    this.orderUid = args.orderUid;
-    this.type = TransactionInfoType.SwapOrder;
-    this.status = args.status;
-    this.orderKind = args.orderKind;
-    this.sellToken = args.sellToken;
-    this.buyToken = args.buyToken;
-    this.expiresTimestamp = args.expiresTimestamp;
-    this.filledPercentage = args.filledPercentage;
-    this.explorerUrl = args.explorerUrl;
-  }
-}
-
-@ApiExtraModels(TokenInfo)
-export class FulfilledSwapOrderTransactionInfo extends SwapOrderTransactionInfo {
-  @ApiProperty({ enum: ['fulfilled'] })
-  override status: 'fulfilled';
+  sellAmount: string;
 
-  @ApiPropertyOptional({
-    type: String,
-    nullable: true,
-    description:
-      'The amount of fees paid for this order in the format of "$feeAmount $tokenSymbol"',
+  @ApiProperty({
+    description: 'The buy token raw amount (no decimals)',
   })
-  feeLabel: string | null;
+  buyAmount: string;
 
   @ApiProperty({
-    description:
-      'The execution price label is in the format of "1 $sellTokenSymbol = $ratio $buyTokenSymbol"',
+    description: 'The executed sell token raw amount (no decimals)',
   })
-  executionPriceLabel: string;
+  executedSellAmount: string;
 
   @ApiProperty({
-    description:
-      'The (averaged) surplus for this order in the format of "$surplusAmount $tokenSymbol"',
+    description: 'The executed buy token raw amount (no decimals)',
   })
-  surplusLabel: string;
+  executedBuyAmount: string;
 
-  constructor(args: {
-    orderUid: string;
-    orderKind: 'buy' | 'sell';
-    sellToken: TokenInfo;
-    buyToken: TokenInfo;
-    expiresTimestamp: number;
-    feeLabel: string | null;
-    executionPriceLabel: string;
-    surplusLabel: string;
-    filledPercentage: string;
-    explorerUrl: URL;
-  }) {
-    super({ ...args, status: 'fulfilled' });
-    this.status = 'fulfilled';
-    this.feeLabel = args.feeLabel;
-    this.executionPriceLabel = args.executionPriceLabel;
-    this.surplusLabel = args.surplusLabel;
-  }
-}
+  @ApiProperty({ description: 'The sell token of the order' })
+  sellToken: TokenInfo;
+
+  @ApiProperty({ description: 'The buy token of the order' })
+  buyToken: TokenInfo;
```

Comment:
What do you think about putting these in nested structures? We could reuse the types, e.g.

```ts
type OrderInfo = {
  amount: string;
  executedAmount: string;
  token: TokenInfo
}

type SwapOrderTransactionInfo = {
  buy: OrderInfo;
  sell: OrderInfo;
  ...
}
```

Reply:
@compojoom @usame-algan any opinion on this? I've tried to match it closer to what the CoW Swap API returns. Do you prefer this alternative (involves more nesting: Root -> OrderInfo -> TokenInfo)

Reply:
I think both would work. Since there are always exactly two tokens in a swap, not nesting it is fine as it won't need to be extended in the future. I would be in favour of leaving as is since its less work.